converters/toZigbee.js

```diff
@@ -2364,6 +2364,86 @@ const converters = {
+            }
+            if (value.brightness) {
+                payload.brightness = value.brightness;
+            }
+            if (typeof value === 'number') {
+                payload.brightness = value;
+            }
+            // if key is color -> make sure to switch to rgb mode
+            await entity.command('lightingColorCtrl', 'tuyaRgbMode', {enable: 1}, {}, {disableDefaultResponse: true});
+            await entity.command('lightingColorCtrl', 'tuyaMoveToHueAndSaturationBrightness', payload, {disableDefaultResponse: true});
+            // transtime cannot be set on these devices. They seem to have a default one of about 1500ms!
+            return {state: {color_temp: value}, readAfterWriteTime: payload.transtime * 100};
+        },
+        convertGet: async (entity, key, meta) => {
+            if (key === 'color') {
+                await entity.read('lightingColorCtrl', ['currentHue', 'currentSaturation', 'tuyaBrightness', 'tuyaMode']);
+            }
+        },
+    },
+    tuya_brightness_control: {
+        key: ['brightness'],
+        convertSet: async (entity, key, value, meta) => {
+            if (meta.state.tuyaMode === 0) {
+                await entity.command('genLevelCtrl',
+                    'moveToLevel',
+                    {transtime: 0, level: value},
+                    {disableResponse: true, disableDefaultResponse: true});
+                await entity.command('lightingColorCtrl', 'tuyaRgbMode', {enable: 0}, {}, {disableDefaultResponse: true});
+            } else {
+                const payload = {
+                    transtime: 0,
+                    hue: Math.round((meta.state.color.h * 254) / 360),
+                    saturation: Math.round(meta.state.color.s * 2.54),
+                    brightness: value,
+                };
+                // if key is color -> make sure to switch to rgb mode
+                await entity.command('lightingColorCtrl', 'tuyaRgbMode', {enable: 1}, {}, {disableDefaultResponse: true});
+                await entity.command('lightingColorCtrl', 'tuyaMoveToHueAndSaturationBrightness', payload, {disableDefaultResponse: true});
+            }
+        },
+    },
+    tuya_light_colortemp: {
+        key: ['color_temp'],
+        convertSet: async (entity, key, value, meta) => {
+            // Mapping from
+            // Homeassistant: Coldwhite 153 -> 500 Warmwight
+            // Warmwhite 0 -> 255 Coldwhite
+            //
+            value = Number(value);
+            const mappedValue = Math.round(-0.734 * value + 367);
+            const payload = {colortemp: mappedValue, transtime: 0};
+            // disable tuya rgb mode
+            await entity.command('lightingColorCtrl', 'tuyaRgbMode', {enable: 0}, {}, {disableDefaultResponse: true});
+            await entity.command('lightingColorCtrl', 'moveToColorTemp', {
+                ...payload,
+                brightness: meta.state.brightness,
+            }, getOptions(meta.mapped, entity));
```

Comment:
Please also return the new state here (https://github.com/Koenkk/zigbee-herdsman-converters/pull/1661/files#r504752251)